### PR TITLE
Add Temporal filter

### LINKED ISSUE #509: Document temporal filter
https://github.com/risingwavelabs/rfcs/blob/main/rfcs/0018-temporal-filter.md

### PR DESCRIPTION
**Pull request by bernscode**
Add Temporal filter.

## Info
- **Description**: 
Add Temporal filter doc.

- **Preview**: 
https://pr-627.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-pattern-temporal-filter/

- **Related code PR**: 
N/A

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/509

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/sql/syntax/sql-pattern-temporal-filter.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Temporal filter is a feature that allows you to filter data based on time intervals, and it's used to retrieve data within a specific time range. A temporal filter will enable you to filter data based on a particular time, such as the current time, a specific date, or a range of dates. By using a temporal filter, you can ensure that your queries only return data relevant to the period you are interested in, making your data analysis more accurate and efficient.
+
+
+The following example query returns all rows from the `sales` table where the `sale_date` column plus one week is greater than the current date and time. In other words, it will return all sales records within the past week.
+
+```sql
+SELECT * 
+FROM sales 
+WHERE sale_date + INTERVAL '1 week' > NOW();
```
Comment (fuyufjh) [inline docs/sql/syntax/sql-pattern-temporal-filter.md]: There are equivalent for sure, but this looks more intuitive to me.
 
```suggestion
WHERE sale_date > NOW() - INTERVAL '1 week';
```

File: docs/sql/syntax/sql-pattern-temporal-filter.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+```
+
+The temporal filter in this query is `sale_date + INTERVAL '1 week' > NOW()`. It filters the rows based on the `sale_date` column and checks if it is within one week of the current time or `NOW()`.
```
Comment (fuyufjh) [inline docs/sql/syntax/sql-pattern-temporal-filter.md]: ditto.

```suggestion
The temporal filter in this query is `sale_date > NOW() - INTERVAL '1 week'`. It filters the rows based on the `sale_date` column and checks if it is within one week of the current time or `NOW()`.
```

**Review by fuyufjh — APPROVED** [review]
LGTM for the rest

File: docs/sql/syntax/sql-pattern-temporal-filter.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+id: sql-pattern-temporal-filter
+slug: /sql-pattern-temporal-filter
+title: Temporal filter
+---
+
+Temporal filter is a feature that allows you to filter data based on time intervals, and it's used to retrieve data within a specific time range. A temporal filter will enable you to filter data based on a particular time, such as the current time, a specific date, or a range of dates. By using a temporal filter, you can ensure that your queries only return data relevant to the period you are interested in, making your data analysis more accurate and efficient.
```
Comment (hengm3467) [inline docs/sql/syntax/sql-pattern-temporal-filter.md]: @bernscode  As a general practice, we don't recommend using "feature", as it is too generic and can mean anything (a single button or a very comprehensive set of functions). Just describing its function would be fine. Also, we may want to use the plural form for both temporal filters and dynamic filters.

File: docs/sql/syntax/sql-pattern-temporal-filter.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+
+The following example query returns all rows from the `sales` table where the `sale_date` column plus one week is greater than the current date and time. In other words, it will return all sales records within the past week.
```
Comment (hengm3467) [inline docs/sql/syntax/sql-pattern-temporal-filter.md]: Please remove "example" here.

File: docs/sql/syntax/sql-pattern-temporal-filter.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+```sql
+SELECT * 
+FROM sales 
+WHERE sale_date > NOW() - INTERVAL '1 week';
+```
+
+The temporal filter in this query is `sale_date > NOW() - INTERVAL '1 week'`. It filters the rows based on the `sale_date` column and checks if it is within one week of the current time or `NOW()`.
+
+
+The following example query returns all rows from the `user_sessions` table where the sum of the `last_active` timestamp and double the `session_timeout` duration is greater than the current timestamp, indicating active user sessions. This query could be used to clean up old user sessions from the database by deleting any rows that no longer satisfy the condition.
```
Comment (hengm3467) [inline docs/sql/syntax/sql-pattern-temporal-filter.md]: Ditto.

File: docs/sql/syntax/sql-pattern-temporal-filter.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+```sql
+SELECT * 
+FROM user_sessions 
+WHERE last_active + session_timeout * 2 > NOW();
+```
+
+The temporal filter in this query is in the `WHERE` clause, where it checks whether the timestamp of the last activity plus twice the session timeout is greater than the current time or `NOW()`, indicating that the session is still active.
```
Comment (hengm3467) [inline docs/sql/syntax/sql-pattern-temporal-filter.md]: Can we break it into shorter sentences?

**Review by hengm3467 — COMMENTED** [review]
Otherwise looks good to me. After implementing the comments, you can merge the PR. Thanks. Good work!

File: docs/sql/syntax/sql-pattern-temporal-filter.md (hunk 3 above)
Comment (bernscode) [reply docs/sql/syntax/sql-pattern-temporal-filter.md]: @hengm3467 Good point! I've made the update on this and the dynamic filter doc. Thanks!

**Pull request merged**